cmd/fleet/cron.go

```diff
@@ -847,5 +854,14 @@ func startAppleMDMDEPProfileAssigner(
 			}
 			return syncer.Run(ctx)
 		}),
-	).Start()
+	)
+	s.Start()
+
+	return s, nil
+}
+
+func shutdownCronSchedulesService(ctx context.Context, ds fleet.Datastore, logger kitlog.Logger, instanceID string) {
```

Comment:
So what happens in the case where a fleet instance is halfway through running a cron and the status is in pending, and it receives a SIGKIL? If I recall correctly, the other instances will see the pending and skip ahead forever... am I missing something?

Reply:
This would be a reason to get the lock first, so that if it expired, we can simply ignore the pending... but maybe I'm missing some other part that covers this case, so I'll put a pin in these thoughts.

Reply:
Where's this code? I can't find it.

Reply:
https://github.com/fleetdm/fleet/pull/8747/files#diff-7f3c850c0fd5fdacd05143fa0379ed590211abb6ea9019aec42eeaefe2f0f81cR64

Reply:
It's not showing in the diff. You'll find it just below the `UpdateAllCronStatsForInstance` [method](https://github.com/fleetdm/fleet/pull/8747/files#diff-7f3c850c0fd5fdacd05143fa0379ed590211abb6ea9019aec42eeaefe2f0f81cR54).

Reply:
Gotcha, makes sense!